app/models/content_consistency_checker.rb

```diff
@@ -0,0 +1,111 @@
+require 'gds_api/content_store'
+
+class ContentConsistencyChecker
+  attr_reader :content_id, :locale
+
+  def initialize(content_id, locale = "en")
+    @content_id = content_id
+    @locale = locale
+    @errors = []
+  end
+
+  def call
+    return @errors unless document
+
+    check_edition(draft, draft_content_store) if draft
+    check_edition(live, live_content_store) if live
+
+    @errors
+  end
+
+private
+
+  def item_from_content_store(path, content_store)
+    begin
+      content_store.content_item(path).parsed_content
+    rescue GdsApi::ContentStore::ItemNotFound, GdsApi::HTTPGone, GdsApi::HTTPForbidden
+      nil
+    end
+  end
+
+  def check_edition_existence(edition, content_item)
+    if edition.gone? && content_item
+      @errors << "Content exists in the content store."
+    elsif content_item.nil?
+      @errors << "Content is missing from the content store."
+    end
+  end
+
+  def check_edition(edition, content_store)
+    redirects = edition.redirects
+
+    redirects.each do |redirect|
+      path = redirect[:path]
+
+      content_item = item_from_content_store(path, content_store)
+
+      check_edition_existence(edition, content_item)
+
+      next unless content_item
+
+      if content_item["schema_name"] != "redirect"
+        @errors << "Content item at #{path} not registered as a redirect."
+      end
+    end
+
+    routes = edition.routes
+
+    routes.each do |route|
+      path = route[:path]
+
+      content_item = item_from_content_store(path, content_store)
+
+      check_edition_existence(edition, content_item)
+
+      next unless content_item
+
+      if content_item["schema_name"] == "redirect"
+        @errors << "Content item at #{path} is registered as a redirect."
+      end
+
+      if content_item["rendering_app"] != edition.rendering_app
+        @errors << "Rendering app " \
+                   "(#{content_item['rendering_app']}) does not " \
+                   "match item rendering app (#{edition.rendering_app})."
+      end
+
+      if content_item["schema_name"] != edition.schema_name
+        @errors << "Schema name (#{content_item['schema_name']}) does " \
+                   "not match item schema name (#{edition.schema_name})."
+      end
+
+      if content_item["document_type"] != edition.document_type
+        @errors << "Document type (#{content_item['document_type']}) does " \
+                   "not match item document type (#{edition.document_type})."
+      end
+    end
+  end
+
+  def live
+    document.live
+  end
+
+  def draft
+    document.draft
+  end
+
+  def document
+    @document ||= Document.find_by!(content_id: content_id, locale: locale)
```

Comment:
We could probably just have a `find_by` rather than the banged equivalent here and deal with a document of nil in the `call` method. Save a place where we mutate `errors`